src/ranking.jl

```diff
@@ -171,3 +171,174 @@ Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; sortkwargs...) =
     _rank(_tiedrank!, x, Float64; sortkwargs...)
+
+"""
+    quantilerank(v, value; method=:inc)
+
+Compute the quantile(s)-position in [0, 1] interval of a `value` relative to a collection 
+`v`, e.g. a quantile rank of x means that (x*100)% of the elements in `v` are lesser 
+(strict) or lesser-equal (weak) the given `value`. 
+
+The function gets the counts of elements of `v` that are less than `value` (`count_less`), 
+elements of `v` that are equal to `value` (`count_equal`) and the length of `v` (`n`). 
+It also extracts the highest value of elements below `value` (`last_less`) and the lowest 
+value of elements above `value` (`less_greater`). Using them, different methods and 
+definitions of the `quantilerank` are obtained by changing the `method` keyword argument: 
+
+`:inc` (default) - It calculates a value in the range 0 to 1 inclusive. 
+If `value ∈ v`, it returns `count_less / (n - 1)`, if not, apply interpolation based on 
+def. 7 in Hyndman and Fan (1996). 
```

Comment:
This method is the inverse of `quantile`, right? If we can confirm this empirically on a few examples that cover all cases (e.g. repeated values, all values unique, all values equal, etc.), then I'd mention it even if we don't have the formal proof.

Reply:
> This method is the inverse of quantile, right? 

I can't confirm because I didn't conclude that. In the past, you said that to me and then I include in the definition of `:inc` and `:exc`. And recently, I remove that during a question of bkamins. But I could include again, if you prefer. 

![image](https://user-images.githubusercontent.com/44294500/151680115-adc93d07-ed40-41f3-9234-01d664390033.png)

Reply:
> I can't confirm because I didn't conclude that. 

Do you mean that you found cases where this didn't work or just that you don't have the mathematical proof?

Reply:
I don't have the mathematical proof

Reply:
This method is not inverse of `quantile` because `quantile` is a continuous function while `quantilerank` is not.
(but they are the same for points that are not in the data set)

Also `quantilerank` requires the collection to have at least 2 elements while `quantile` accepts 1-element collections. So they differ here also.

Reply:
A sample example:
```
julia> quantilerank([2.0, 2.0], 2.0)
0.0

julia> quantile([2.0, 2.0], 1.0)
2.0
```

Reply:
Yes this is a bit tricky. But even if this is only the case in particular conditions, it could still be interesting to mention as it makes what the function does much clearer. Do you think we could find a phrasing which would be correct yet relatively short?

Reply:
we could say that `quantilerank` is the lower semi-continuous inverse of `quantile` with default parameters (slightly abusing mathematical notation but I think that this description is clear).